src/Strategy/Python/Poetry.hs

```diff
@@ -46,8 +46,10 @@ discover dir = context "Poetry" $ do
   pure (map mkProject projects)
 
 -- | Poetry build backend identifier required in [pyproject.toml](https://python-poetry.org/docs/pyproject/#poetry-and-pep-517).
-poetryBuildBackendIdentifier :: Text
-poetryBuildBackendIdentifier = "poetry.core.masonry.api"
+usesPoetryBackend :: Text -> Bool
+usesPoetryBackend backend =
+  backend == "poetry.core.masonry.api" -- For poetry versions >=1.1.0a1 (released 2020)
+    || backend == "poetry.masonry.api"
```

Comment:
Is there any link to documentation or tool source code for this magic string? If so, please add as a comment here.